src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -12,81 +23,101 @@
-        new PersonBuilder()
-            .name("Sami")
-            .gitHubID("Sami-2000")
-            .description("Sami is fun.")
-            .interestTags(SamiSkills)
-            .projectIDs(projectIDs)
-            .buildPerson();
-    people.add(Sami);
-
-    ArrayList<String> RichiSkills = new ArrayList<String>();
-    RichiSkills.add("Minecraft boss");
-    RichiSkills.add("React wizard");
-    PersonObject Richi =
-        new PersonBuilder()
-            .name("Richi")
-            .gitHubID("Richie78321")
-            .description("Richi is cool.")
-            .interestTags(RichiSkills)
-            .projectIDs(projectIDs)
-            .buildPerson();
-    people.add(Richi);
+    datastore = DatastoreServiceFactory.getDatastoreService();
   }
 
   @Override
-  // get a specific user. return null if not found. TODO: User Validation
+  // Get a specific user. return null if not found. TODO: User Validation
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userString = request.getParameter("user");
-
-    PersonObject result =
-        people.stream()
-            .filter(person -> userString.equals(person.getGitHubID()))
-            .findAny()
-            .orElse(null);
-    if (result == null) {
-      sendRawTextError(
-          response, HttpServletResponse.SC_BAD_REQUEST, "invalid user or user does not exist");
+    String userGithub = request.getParameter("githubID");
+    Key userKey = KeyFactory.createKey(PersonObject.ENTITY_NAME, userGithub);
+
+    PersonObject userObject;
+
+    try {
+      userObject = toPersonObject(datastore.get(userKey));
+    } catch (Exception e) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Person does not exist in Datastore");
+      userObject = null;
     }
 
-    String jsonPerson = new Gson().toJson(result);
+    String jsonPerson = new Gson().toJson(userObject);
     response.setContentType("application/json;");
     response.getWriter().println(jsonPerson);
   }
 
-  /** Sends an error to the client as raw text instead of the default HTML page. */
-  private void sendRawTextError(HttpServletResponse response, int errorCode, String errorMsg)
-      throws IOException {
-    response.setStatus(errorCode);
-    response.setContentType("text/html;");
-    response.getWriter().println(errorMsg);
+  @Override
+  // Send a user to datastore. Update the current information about the user if one exists.
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String userGitHubID = request.getParameter("gitHubID");
+    if (userGitHubID.isBlank()) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid ID");
+      return;
+    }
+    ArrayList<String> userTags =
+        (ArrayList<String>) Arrays.asList(request.getParameterValues("tags"));
+
+    sendData(new PersonBuilder().gitHubID(userGitHubID).interestTags(userTags).buildPerson());
```

Comment:
Is sendData a servlet function? Sorry for the dumb question, don't see it defined so I'm curious.

Reply:
good catch!